cmd/dra-example-kubeletplugin/cdi.go

```diff
@@ -89,18 +91,37 @@ func (cdi *CDIHandler) CreateClaimSpecFile(claimUID string, devices PreparedDevi
 		Devices: []cdispec.Device{},
 	}
 
-	gpuIdx := 0
-	for _, device := range devices {
+	for i, device := range devices {
+		envs := []string{
+			fmt.Sprintf("GPU_DEVICE_%d=%s", i, device.DeviceName),
+		}
+
+		if device.Config.Sharing.IsTimeSlicing() {
+			tsconfig, err := device.Config.Sharing.GetTimeSlicingConfig()
+			if err != nil {
+				return fmt.Errorf("unable to get time slicing config for device %v: %v", device.DeviceName, err)
+			}
+			envs = append(envs, fmt.Sprintf("GPU_SHARING_STRATEGY=%s", configapi.TimeSlicingStrategy))
+			envs = append(envs, fmt.Sprintf("GPU_TIMESLICE_DURATION=%v", tsconfig.Duration))
+		}
+
+		if device.Config.Sharing.IsSpacePartitioning() {
+			spconfig, err := device.Config.Sharing.GetSpacePartitioningConfig()
+			if err != nil {
+				return fmt.Errorf("unable to get space partitioning config for device %v: %v", device.DeviceName, err)
+			}
+			envs = append(envs, fmt.Sprintf("GPU_SHARING_STRATEGY=%s", configapi.SpacePartitioningStrategy))
```

Comment:
why not just use `config.Sharing.Strategy` directly in both cases?

Reply:
Will do in next iteration.

Reply:
Updated.